public/app/plugins/panel/timeseries/utils.ts

```diff
@@ -173,3 +179,29 @@ export function regenerateLinksSupplier(
 
   return alignedDataFrame;
 }
+
+export function prepareTimeSeriesLong(series: DataFrame[]): DataFrame[] {
+  // Transform each dataframe of the series
+  // to handle different field names in different frames
+  return series.reduce((acc: DataFrame[], dataFrame: DataFrame) => {
+    // these could be different in each frame
+    const stringFields = dataFrame.fields.filter((field) => field.type === FieldType.string).map((field) => field.name);
+
+    // transform one dataFrame at a time and concat into DataFrame[]
+    const transformedSeries = partitionByValuesTransformer.transformer(
+      {
+        fields: stringFields,
+        naming: {
```

Comment:
i think this can all be skipped? they're just defaults. and most don't have any effect when `asLabels: true` (the default).

https://github.com/grafana/grafana/blob/ca4cd8550479cb3e07f3bf1a4cbf2fba64af07e7/public/app/features/transformers/partitionByValues/partitionByValues.ts#L20

Reply:
Cool, I can confirm this. I'll make that change. Thanks!